File: wsh.jscript/g01.array.js
// create empty list
var nicknames = []

// resize array and insert element by index
nicknames[0] = "bob"
nicknames[1] = "ed"
nicknames[2] = "steve"
nicknames[3] = "ralph"
nicknames[4] = "joe"
nicknames[5] = "deb"
nicknames[6] = "kate"

WScript.echo("The total nicknames are: " + nicknames.length);
WScript.echo("The nicknames are: " + nicknames.join(", "));

/***********************************************
 * Array.join(sep) method - returns string given array and separator
 *  Helper method to help enumerate a list easily 
 ***********************************************/
 Array.prototype.join = function(sep)  {
    result = this[0]; // fetch 1st item from array

    for (i = 1; i < array.length; i++) 
        result += sep + this[i];  // append separator & array item
 
    return result;     // return result
 }

File: gen_scripts/python2/g01.array.py
#!/usr/bin/env python2
nicknames = []           # create an empty list

# populate list one item at a time 
nicknames.append("bob")
nicknames.append("ed")
nicknames.append("steve")
nicknames.append("ralph")
nicknames.append("joe")
nicknames.append("deb")
nicknames.append("kate")

# output length and all values
print "The total nicknames are: %d" % len(nicknames)
print "The nicknames are: " + ", ".join(nicknames)
